x-pack/plugins/enterprise_search/public/applications/app_search/components/schema/components/truncated_engines_list.tsx

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { Fragment } from 'react';
+
+import { EuiText, EuiToolTip, EuiButtonEmpty } from '@elastic/eui';
+
+import { EuiLinkTo } from '../../../../shared/react_router_helpers';
+import { ENGINE_SCHEMA_PATH } from '../../../routes';
+import { generateEncodedPath } from '../../../utils/encode_path_params';
+
+interface Props {
+  engines?: string[];
+  cutoff?: number;
+}
+
+export const TruncatedEnginesList: React.FC<Props> = ({ engines, cutoff = 3 }) => {
+  if (!engines?.length) return null;
+
+  const displayedEngines = engines.slice(0, cutoff);
+  const hiddenEngines = engines.slice(cutoff);
+  const SEPARATOR = ', ';
```

Comment:
I think the difficulty here is that I'm using `SEPARATOR` in various places and not just as a joined array (e.g. conditionally if the tooltip truncation kicks in, in a `.map` because we're using `EuiLinkTo` for the engine names, etc. 🤔

To be honest, I think a better approach _might_ be to use something like `<EuiBadge>` or some other EUI list component to list out engines in any case... Maybe just copy what we do for analytics tags:

<img width="300" alt="" src="https://user-images.githubusercontent.com/549407/118324078-6a930680-b4b6-11eb-840f-7b24e8a1080a.png">

Our tag truncation component ([link](https://github.com/elastic/kibana/blob/master/x-pack/plugins/enterprise_search/public/applications/app_search/components/analytics/components/analytics_tables/tags.tsx#L36-L64)) also uses `and X more` text instead of a "+X" copy. We avoid all the i18n shenanigans with commas AND the + with that approach 🤔 

@daveyholler what do you think?... would engine names in badges be weird?